### DISCUSSION TITLE
Is 1.1 the latest version of SceneViewer?

### DISCUSSION THREAD

**Discussion by DerrickBarra** [google/model-viewer#2409] [Q&A] [answered]
Our team is loading scene-viewer via a generated link in Javascript...

`var intent = "intent://arvr.google.com/scene-viewer/1.1?file=";
		var scheme = "#Intent;scheme=https;package=com.google.android.googlequicksearchbox;action=android.intent.action.VIEW;S.browser_fallback_url=";
		var fallBack = "https://developers.google.com/ar;end;";
		var newLink = intent.concat(glbLink).concat('&mode=ar_only').concat(scheme).concat(fallBack);
		window.open(newLink);`

Is 1.1 the latest version of scene-viewer? Where can I find a list of the version numbers? Is there a way we can ensure we're always loading the latest version?

We're trying to make sure our rendering always looks as good as the SceneViewer links generated by the QR code with the [Model Viewer editor](https://modelviewer.dev/editor/), which in our experience has improved rendering than our generated scene-viewer links in javascript.

In the snapshot below, you can see what the rendering looks like in Model-Viewer editor (and in the Scene-Viewer that the Model-Viewer tool generates) vs what it looks like in the older Scene-Viewer tool & in our code.

![image](https://user-images.githubusercontent.com/20777145/119013563-1d56df00-b965-11eb-9979-81b4d11a0803.png)

**Comment by elalish** [comment]
@tpsiaki @timmmeh Any ideas?

**Comment by felipe-chamas** [comment]
> Is 1.1 the latest version of scene-viewer? Where can I find a list of the version numbers? Is there a way we can ensure we're always loading the latest version?

I believe it's possible to use `1.2`. Although I have never seen any changelog about SceneViewer, the documentation does not show all flags and there is nowhere to be notified about updates... Personally I don't know what's the difference between versions apart from Intent compatibility on version `1.1`.
It would be very nice to have a place of `SceneViewer` discussions, questions and updates, just like we have it here with `model-viewer` (and I'm not asking for an open-source repo). I feel like most devs find this is the better place to ask, even though it is not efficient.

**Reply by elalish** [reply]
Well, MV and SV have recently joined forces org-wise, so this is actually more efficient than it used to be.

**Comment by tpsiaki** [comment, marked as answer]
Scene Viewer intent versioning lets you require a minimum feature set when launching Scene Viewer. The idea is that if you launch with a 1.2 intent then you're guaranteed that if the intent succeeds you have the 1.2 features, and otherwise you'll fallback to whatever you specified as the fallback url.

There's no difference in the actual Scene Viewer experience you'll get based on the version you pass, it only controls whether your intent will launch on (very) old versions of Scene Viewer.

At this point the vast majority of Scene Viewer users are on versions that support 1.2 so realistically it should make very little difference.

For reference, we added the 1.2 version of the Scene Viewer intent when we added Draco compression support.

In the future I do expect that we'll add additional version numbers when new non-backwards-compatible features are added. For example, we will likely add a 1.3 version for basis texture compression support soon. In this case, though, you'd only want to update to using the 1.3 intent if you actually required basis support - otherwise it would unnecessarily fallback when older versions of Scene Viewer are installed.

**Reply by DerrickBarra** [reply]
Thanks @tpsiaki , Looking forward to 1.3, we use .ktx2 textures with a .basis payload, and right now we have to generate models without those textures for scene-viewer compatability, good to know that won't be needed in the near future.

I wonder why then the rendering issue I brought up between the [web-based scene-viewer (and my code)](https://arvr.google.com/scene-viewer-preview) and the newer [model-viewer](https://modelviewer.dev/editor/) that can launch scene-viewer is happening?

Here's a[ link to the model ](https://bxr-models-v2.s3.us-east-2.amazonaws.com/8125cdab4f7f49588e5e81fac21ee652/gltf/ArHigh/8125cdab4f7f49588e5e81fac21ee652.glb)we're testing that produced those differences in rendering...
